internal/dinosaur/pkg/services/quota/ams_quota_service_test.go

```diff
@@ -462,10 +506,169 @@ func Test_AMSReserveQuota(t *testing.T) {
+					GetQuotaCostsForProductFunc: func(organizationID, resourceName, product string) ([]*v1.QuotaCost, error) {
+						rrbq1 := v1.NewRelatedResource().BillingModel(string(v1.BillingModelMarketplace)).Product(string(ocm.RHACSTrialProduct)).ResourceName(resourceName).Cost(0)
+						qcb1, err := v1.NewQuotaCost().Allowed(0).Consumed(2).OrganizationID(organizationID).RelatedResources(rrbq1).Build()
+						if err != nil {
+							panic("unexpected error")
+						}
+						return []*v1.QuotaCost{qcb1}, nil
+					},
+					GetCustomerCloudAccountsFunc: func(externalID string, quotaIDs []string) ([]*v1.CloudAccount, error) {
+						cloudAccount, _ := v1.NewCloudAccount().
+							CloudAccountID("cloudAccountID").
+							CloudProviderID("cloudProviderID").
+							Build()
+						return []*v1.CloudAccount{
+							cloudAccount,
+						}, nil
+					},
 				},
 			},
 			wantErr: true,
 		},
+		{
+			name: "cloud account id in request does not match cloud_accounts response",
+			args: args{
+				"12231",
+				"testUser",
+				"different cloudAccountID",
+			},
+			fields: fields{
+				ocmClient: &ocm.ClientMock{
+					ClusterAuthorizationFunc: func(cb *v1.ClusterAuthorizationRequest) (*v1.ClusterAuthorizationResponse, error) {
+						sub := v1.SubscriptionBuilder{}
+						sub.ID("1234")
+						sub.Status("Active")
+						ca, _ := v1.NewClusterAuthorizationResponse().Allowed(true).Subscription(&sub).Build()
+						return ca, nil
+					},
+					GetOrganisationIDFromExternalIDFunc: func(externalId string) (string, error) {
+						return fmt.Sprintf("fake-org-id-%s", externalId), nil
+					},
+					GetQuotaCostsForProductFunc: func(organizationID, resourceName, product string) ([]*v1.QuotaCost, error) {
+						rrbq1 := v1.NewRelatedResource().BillingModel(string(v1.BillingModelMarketplace)).Product(string(ocm.RHACSTrialProduct)).ResourceName(resourceName).Cost(0)
+						qcb1, err := v1.NewQuotaCost().Allowed(0).Consumed(2).OrganizationID(organizationID).RelatedResources(rrbq1).Build()
+						if err != nil {
+							panic("unexpected error")
+						}
+						return []*v1.QuotaCost{qcb1}, nil
+					},
+					GetCustomerCloudAccountsFunc: func(externalID string, quotaIDs []string) ([]*v1.CloudAccount, error) {
+						cloudAccount, _ := v1.NewCloudAccount().
+							CloudAccountID("cloudAccountID").
+							CloudProviderID("cloudProviderID").
+							Build()
+						return []*v1.CloudAccount{
+							cloudAccount,
+						}, nil
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "cloud account matches cloud_accounts response",
+			args: args{
+				"12231",
+				"testUser",
+				"cloudAccountID",
+			},
+			fields: fields{
+				ocmClient: &ocm.ClientMock{
+					ClusterAuthorizationFunc: func(cb *v1.ClusterAuthorizationRequest) (*v1.ClusterAuthorizationResponse, error) {
+						sub := v1.SubscriptionBuilder{}
+						sub.ID("1234")
+						sub.Status("Active")
+						ca, _ := v1.NewClusterAuthorizationResponse().Allowed(true).Subscription(&sub).Build()
```

Comment:
Can you rename `ca` to `cloudAccount`? ca is an abbreviation used to often for certificate authority.

```suggestion
						cloudAccount, _ := v1.NewClusterAuthorizationResponse().Allowed(true).Subscription(&sub).Build()
```

Reply:
Here `ca` stands for `clusterAuthorization` - should we change to `cloudAuthorizationResponse`?

Reply:
Good suggestion, +1 for `cloudAuthorizationResp`